### PR TITLE
Feature/qr code reader

### PR DESCRIPTION
**Pull request by mrtcelebi**

### CODE REVIEW

**Review by alihanaktay — APPROVED** [review]

**Review by i-Senku — APPROVED** [review]

File: MobilliumQRCodeReader.podspec [hunk 1]
```diff
@@ -8,35 +8,20 @@
 
 Pod::Spec.new do |s|
   s.name             = 'MobilliumQRCodeReader'
-  s.version          = '0.1.0'
+  s.version          = '1.0.0'
   s.summary          = 'A short description of MobilliumQRCodeReader.'
```
Comment (aslanmehmetsalih) [inline MobilliumQRCodeReader.podspec]: can you type summary

Comment (aslanmehmetsalih) [inline MobilliumQRCodeReader.podspec:12]: can you type summary

File: Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+            if let detector = CIDetector(ofType: CIDetectorTypeQRCode, context: nil, options: [CIDetectorAccuracy: CIDetectorAccuracyHigh]) {
+                let ciImage = CIImage(image: qrcodeImage)!
+                var qrCodeString = ""
+                
+                let features = detector.features(in: ciImage)
+                for feature in features as! [CIQRCodeFeature] {
+                    qrCodeString += feature.messageString!
+                }
+                
+                if qrCodeString == "" {
```
Comment (aslanmehmetsalih) [inline Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift]:
```suggestion
                if qrCodeString.isEmpty {
```

Comment (aslanmehmetsalih) [inline Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift:288]:
```suggestion
                if qrCodeString.isEmpty {
```
```suggestion
                if qrCodeString == "" {
```

File: Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+    public func metadataOutput(_ output: AVCaptureMetadataOutput, didOutput metadataObjects: [AVMetadataObject], from connection: AVCaptureConnection) {
+        captureSession.stopRunning()
+        if let metadataObject = metadataObjects.first {
+            if metadataObject.type == .qr {
+                let readableObject = metadataObject as? AVMetadataMachineReadableCodeObject
+                guard let qrCodeString = readableObject?.stringValue else { return }
+                AudioServicesPlaySystemSound(SystemSoundID(kSystemSoundID_Vibrate))
+                delegate?.qrCodeReader(self, didSuccess: qrCodeString)
+            }
+        }
```
Comment (aslanmehmetsalih) [inline Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift]:
```suggestion
        if let metadataObject = metadataObjects.first,
           metadataObject.type == .qr {
            let readableObject = metadataObject as? AVMetadataMachineReadableCodeObject
            guard let qrCodeString = readableObject?.stringValue else { return }
            AudioServicesPlaySystemSound(SystemSoundID(kSystemSoundID_Vibrate))
            delegate?.qrCodeReader(self, didSuccess: qrCodeString)
        }
```

Comment (aslanmehmetsalih) [inline Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift:269]:
```suggestion
        if let metadataObject = metadataObjects.first,
           metadataObject.type == .qr {
            let readableObject = metadataObject as? AVMetadataMachineReadableCodeObject
            guard let qrCodeString = readableObject?.stringValue else { return }
            AudioServicesPlaySystemSound(SystemSoundID(kSystemSoundID_Vibrate))
            delegate?.qrCodeReader(self, didSuccess: qrCodeString)
        }
```

File: Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+    }
+}
+
+// MARK: - Actions
+extension QRCodeReaderViewController {
+    
+    @objc
+    private func closeButtonTapped() {
+        dismiss(animated: true)
```
Comment (aslanmehmetsalih) [inline Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift]: Can be added **qrCodeReaderDismiss** func to **QRCodeReaderDelegate** and call from here in dismiss closure.

Comment (aslanmehmetsalih) [inline Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift:208]: Can be added **qrCodeReaderDismiss** func to **QRCodeReaderDelegate** and call from here.

File: MobilliumQRCodeReader.podspec [hunk 5]
```diff
@@ -8,35 +8,20 @@
...
   s.description      = <<-DESC
-TODO: Add long description of the pod here.
+  MobilliumQRCodeReader is a customisable qr code reader also you can read qr code from gallery image.
                        DESC
 
   s.homepage         = 'https://github.com/mrtcelebi/MobilliumQRCodeReader'
-  # s.screenshots     = 'www.example.com/screenshots_1', 'www.example.com/screenshots_2'
   s.license          = { :type => 'MIT', :file => 'LICENSE' }
   s.author           = { 'mrtcelebi' => 'mrttcelebi@gmail.com' }
   s.source           = { :git => 'https://github.com/mrtcelebi/MobilliumQRCodeReader.git', :tag => s.version.to_s }
```
Comment (ahmetimirze) [inline MobilliumQRCodeReader.podspec]: source must be 'https://github.com/mobillium/MobilliumQRCodeReader.git'

File: MobilliumQRCodeReader.podspec (hunk 1 above)
Comment (mrtcelebi) [reply MobilliumQRCodeReader.podspec]: summary added

File: Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift (hunk 2 above)
Comment (mrtcelebi) [reply Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift]: refactored

File: Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift (hunk 3 above)
Comment (mrtcelebi) [reply Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift]: refactored

File: Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift (hunk 4 above)
Comment (mrtcelebi) [reply Sources/MobilliumQRCodeReader/Classes/QRCodeReaderViewController.swift]: qrCodeReaderDismiss func added to QRCodeReaderDelegate and dismiss method called with delegate func

File: MobilliumQRCodeReader.podspec (hunk 5 above)
Comment (mrtcelebi) [reply MobilliumQRCodeReader.podspec]: fixed.

**Review by aslanmehmetsalih — APPROVED** [review]

**Pull request merged**